include/onnxruntime/core/graph/rewrite_rule.h

```diff
@@ -54,6 +69,63 @@ class GraphEditor {
     return graph_.Resolve();
   }
 
+  // TODO: The following functions should probably be moved to the Graph and Node classes.
+  // We also need to decide whether we keep the GraphEditor or we pass directly the Graph.
+  // If we keep it, it might be better to put it in a separate file.
+
+  /** Check whether the node has a single input and a single output. */
+  bool IsSingleInSingleOutNode(const Node& node) const {
+    return node.GetInputEdgesCount() == 1 && node.GetOutputEdgesCount() == 1;
+  }
+
+  /** Return the attribute of a Node. */
+  const onnx::AttributeProto* GetNodeAttribute(
+      const Node& node, const std::string& attr_name) const {
+    const auto& attrs = node.GetAttributes();
+    const auto iter = attrs.find(attr_name);
+    return iter == attrs.end() ? nullptr : &iter->second;
+  }
+
+  /** Retrieve the values for a repeated attribute of a node and places them to the values vector. */
+  template <typename T>
+  bool GetRepeatedNodeAttributeValues(
+      const Node& node, std::string attr_name, std::vector<T>& values) const {
+    const auto* attr = GetNodeAttribute(node, attr_name);
+    if (attr) {
+      values = onnx::RetrieveValues<T>(*attr);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /** Remove the given single-input-single-output Node from the Graph. */
+  bool RemoveSingleInSingleOutNode(NodeIndex node_to_remove) {
```

Comment:
This should belong a graph transoformer or rewrite rule instead of GraphEditor. Make sense? otherwise, GraphEditor will become bigger and bigger to contain all kinds of rewriting/editing rules/functions.

Reply:
I would prefer to keep these higher level graph-editing functions to a class like GraphEditor and pass a GraphEditor object whenever we want to transform the graph. Then the GraphTransformers can use a GraphEditor too, instead of manipulating directly the Graph object.

Reply:
why this is not a rewrite rule?

Reply:
This method is now part of the graph_utils static methods.